src/speed_lookup_bindings.cpp

```diff
@@ -177,7 +177,22 @@ NAN_METHOD(SpeedLookup::getRouteSpeeds)
             : Base(callback), datamap{datamap_}, nodeIds(std::move(nodeIds))
         {
         }
-        void Execute() override { result_annotations = datamap->getValues(nodeIds); }
+
+        void Execute() override { 
+
+            if (datamap)
+            {
+                result_annotations = datamap->getValues(nodeIds); 
+            }
+            else
+            {
+                for (std::size_t index = 0; index < nodeIds.size(); index++)
```

Comment:
Note that this should be `nodeIds.size() - 1` - we're returning an array of values for the segments between the nodes, so it's always one less than the number of nodes.